towhee/engine/operator_context.py

```diff
@@ -52,13 +52,19 @@ def __init__(
         dataframes: Dict[str, DataFrame]
     ):
         self._repr = op_repr
-
-        inputs = list({dataframes[input['df']] for input in op_repr.inputs})
         iter_type = op_repr.iter_info['type']
-        inputs_index = dict((item['name'], item['col'])
-                            for item in op_repr.inputs)
-        self.inputs = inputs
-        self._reader = create_reader(inputs, iter_type, inputs_index)
+
+        reader_inputs = {}
+        for x in op_repr.inputs:
+            # If a dataframe is already added, append the new columns.
+            if x['df'] in reader_inputs.keys():
+                reader_inputs[x['df']]['cols'].append((x['name'], x['col']))
+            else:
+                reader_inputs[x['df']] = {}
+                reader_inputs[x['df']]['df'] = dataframes[x['df']]
+                reader_inputs[x['df']]['cols'] = [(x['name'], x['col'])]
+
+        self._reader = create_reader(reader_inputs, iter_type)
```

Comment:
@junjiejiangjjj How do you feel about this change. I think that moving to a dictionary that stores all the info about the input to be easier to use and adapt. Its structure is like this. 
{'_start_df': 
                    {'cols': [('num1', 0), ('num2', 1), ('num3', 2)],
                     'df': <towhee.dataframe.dataframe.DataFrame object at 0x10f48e8b0>}}

Reply:
> 

`OrderedDict` can keep the order always the same, but can not control the order. If we want df1's row always before df2's row if  their  have same row key, `List` can do this, but `OrderedDict` can't.